cognify/frontends/dspy/connector.py

```diff
@@ -41,48 +41,62 @@ def cognify_predictor(
 
         if not isinstance(dspy_predictor, dspy.Predict):
             warnings.warn(
-                "Original module is not a `Predict`. This may result in lossy translation",
+                "Original module is NOT a `dspy.Predict`. This may result in lossy translation",
                 UserWarning,
             )
 
         if isinstance(dspy_predictor, dspy.Retrieve):
             warnings.warn(
-                "Original module is a `Retrieve`. This will be ignored", UserWarning
+                "Original module is a `dspy.Retrieve`. This will be ignored", UserWarning
             )
             self.ignore_module = True
             return None
 
         # initialize cog lm
-        system_prompt = prepare_instructions(dspy_predictor.signature)
         input_names = list(dspy_predictor.signature.input_fields.keys())
         input_variables = [Input(name=input_name) for input_name in input_names]
 
         output_fields = dspy_predictor.signature.output_fields
         if "reasoning" in output_fields:
-            del output_fields["reasoning"]
+            # stripping the reasoning field may crash their workflow, so we warn users instead
             warnings.warn(
-                "Original module contained reasoning. This will be stripped. Add reasoning as a cog instead",
+                f"DSPy module {name} contained reasoning. This may lead to undefined behavior.", 
```

Comment:
this needs more explanation to the user. say something like Cognify performs its own reasoning prompt optimization automatically, which may not work well together with DSPy's; consider removing DSPy's reasoning prompts.